GitHub actions

beqabeqa473 opened the pull request:
for now, this pr enables github actions to build android app and windows packages on every push to master branch
built binaries are uploaded as an artefacts to github storage.

alex19EP commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
```
```suggestion
          java-version: 8
```

similar to: https://github.com/RHVoice/RHVoice/blob/770546edd05dc237f098c9f3867e1702bd521806/.github/workflows/codeql-analysis.yml#L44

alex19EP commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+        with:
+          java-version: 1.8
+      - name: Assemble app debug APK
+        working-directory: ./src/android
```
you can specify this for the whole job.

alex19EP commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+        run: sudo ./gradlew assembleDebug --stacktrace
+
+      - name: Upload app APK
+        uses: actions/upload-artifact@v1
```
```suggestion
        uses: actions/upload-artifact@v2
```

update.

alex19EP commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+        uses: actions/upload-artifact@v1
+        with:
+          name: app-debug
+          path: src/android/RHVoice-core/build/outputs/apk/debug/RHVoice-core-debug.apk
```
if you specified running dir for the whole job you can use Relative path.

beqabeqa473 replied:
No, working-directory is not compatible with actions, so this change is not possible

alex19EP replied:
are you sure?

beqabeqa473 replied:
yes, at least i tried and it didn't work and i've read on github.community that it is not possible to use working-directory in actions

alex19EP replied:
ok.

alex19EP commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+    runs-on: windows-latest
+
+    steps:
+      - uses: actions/checkout@v1
```
```suggestion
      - uses: actions/checkout@v2
```

update.

beqabeqa473 replied:
Done

alex19EP commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
```
```suggestion
        uses: actions/setup-python@v2
```

update.

beqabeqa473 replied:
Done

alex19EP commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+      - name: Install dependencies
+        run: python -m pip install --upgrade pip lxml scons
+      - name: build
+        run: scons
```
```suggestion
        run: scons -j 2
```

let's use 2 CPUs. that's faster.

beqabeqa473 replied:
Done

alex19EP commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+        run: scons
+
+      - name: Upload packages archive
+        uses: actions/upload-artifact@v1
```
```suggestion
        uses: actions/upload-artifact@v2
```

update.

beqabeqa473 replied:
Done

beqabeqa473 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+          java-version: 1.8
+      - name: Assemble app debug APK
+        working-directory: ./src/android
+        run: sudo ./gradlew assembleDebug --stacktrace
```
For the time i wrote workflow gradle on github vms was installed systemwide and there was no way to download gradle dependencies.
Now fixed and sudo is not required.

beqabeqa473 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+name: RHVoice-CI
```
Done

beqabeqa473 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+jobs:
+  androidBuild:
+    name: Build android app
+    runs-on: ubuntu-18.04
```
Done

abitrolly replied:
@alex19EP `latest` Ubuntu according to M$ is 18.04 - see https://github.com/actions/virtual-environments#available-environments so better be explicit about version.

alex19EP replied:
> latest Ubuntu according to M$ is 18.04

yes i know that. and it becomes 20.4 in neer future.

> better be explicit about version.

i don't see a problem with it.

abitrolly replied:
Do you have a prooflink that it will? Upgrading will most likely break builds for many people, so it will not happen.

alex19EP replied:
/actions/virtual-environments/issues/1816

beqabeqa473 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/checkout@v1
```
Done

beqabeqa473 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,48 @@
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
```
Done

abitrolly replied:
GitHub Actions doesn't automatically use latest minor version, it only understands v1 tag literally. What a mess.. )

alex19EP replied:
it's actually good for version binding.

beqabeqa473 commented:
@alex19EP I hope i've addressed al review comments.

is there anything to do for this pr?

alex19EP commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
```
i think we should run this on pr. look at:

https://github.com/RHVoice/RHVoice/blob/770546edd05dc237f098c9f3867e1702bd521806/.github/workflows/codeql-analysis.yml#L3-L8

alex19EP reviewed (changes requested):
rebase on master would be nice.
also, can you add compilation on linux? without uploading artifacts.

thanks.

beqabeqa473 commented:
i thought to include deb/rpm packaging stuff, but for now, i don't
have enough time to make packages.

as for compilation on linux, yes, i can, but what would be a case?
just to make sure it builds?

On 12/17/20, Alexander Epaneshnikov <notifications@github.com> wrote:
> @alex19EP requested changes on this pull request.
>
> rebase on master would be nice.
> also, can you add compilation on linux? without uploading artifacts.
>
> thanks.
>
>> @@ -0,0 +1,51 @@
> +name: CI
> +
> +on:
>
> i think we should run this on pr. look at:
>
> https://github.com/RHVoice/RHVoice/blob/770546edd05dc237f098c9f3867e1702bd521806/.github/workflows/codeql-analysis.yml#L3-L8
>
>> +
> +    steps:
> +      - uses: actions/checkout@v1
> +      - name: Set up JDK 1.8
> +        uses: actions/setup-java@v1
> +        with:
> +          java-version: 1.8
> +      - name: Assemble app debug APK
> +        working-directory: ./src/android
> +        run: sudo ./gradlew assembleDebug --stacktrace
> +
> +      - name: Upload app APK
> +        uses: actions/upload-artifact@v1
> +        with:
> +          name: app-debug
> +          path:
> src/android/RHVoice-core/build/outputs/apk/debug/RHVoice-core-debug.apk
>
> are you sure?
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/RHVoice/RHVoice/pull/131#pullrequestreview-554502456


-- 
with best regards beqa

alex19EP commented:
> as for compilation on linux, yes, i can, but what would be a case?
just to make sure it builds?

exactly.

beqabeqa473 commented:
when ubuntu 20.04 will be latest, we will fix problems if they'll arrive

On 12/17/20, Alexander Epaneshnikov <notifications@github.com> wrote:
> @alex19EP commented on this pull request.
>
>
>
>> @@ -0,0 +1,48 @@
> +name: RHVoice-CI
> +
> +on:
> +  push:
> +    branches:
> +      - 'master'
> +
> +jobs:
> +  androidBuild:
> +    name: Build android app
> +    runs-on: ubuntu-18.04
>
> /actions/virtual-environments/issues/1816
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/RHVoice/RHVoice/pull/131#discussion_r545080683


-- 
with best regards beqa

beqabeqa473 commented:
@alex19EP i think i addressed review comments.

workflow now is triggering on pull requests and building on linux is also in place.

is there anything i can do?

alex19EP reviewed (approved):
i think everything is fine now.
